### PR TITLE
response,extensions: add useful helpers to Authorization and Extension::List

### PR DESCRIPTION
**Pull request by unleashed**
Add helpers for `Authorization` to query and extract its contents, and add a helper to `Extension::List` to easily enable the `list_app_keys` extension.

### CODE REVIEW

File: src/response.rs [hunk 1]
```diff
@@ -57,6 +57,41 @@ pub enum Authorization {
     Error(AuthorizationError),
 }
 
+impl Authorization {
```
Comment (davidor) [inline src/response.rs:60]: no tests?

Comment (unleashed) [reply src/response.rs:60]: :scream: hold my beer

Comment (unleashed) [reply src/response.rs:23]: Ok, I ended up refactoring a bit and adding quite a bit more helpers - in particular a few helpers to help with in-place modification of usage reports, which can become useful for caching. I think we can still add a few more tests, or rather asserts in the existing ones. Can you take a look?

File: src/response/usage_report.rs [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    pub fn current_value(&self) -> u64 {
+        self.current_value
+    }
+
+    pub fn remaining(&self) -> u64 {
+        self.max_value.saturating_sub(self.current_value)
+    }
+
+    pub fn is_limited(&self) -> bool {
+        self.current_value >= self.max_value
```
Comment (davidor) [inline src/response/usage_report.rs:71]: Shouldn't this be `>` ? :thinking:

File: src/response/usage_report.rs [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+    pub fn report(&mut self, hits: u64) -> Result<u64, UsageReportError> {
+        self.current_value = self
+            .current_value
+            .checked_add(hits)
+            .ok_or(UsageReportError::Overflow)?;
+
+        Ok(self.current_value)
+    }
+
+    pub fn reset<V: Into<Option<u64>>>(&mut self, val: V) {
```
Comment (davidor) [inline src/response/usage_report.rs:95]: :tophat:

File: src/response/usage_report.rs (hunk 2 above)
Comment (unleashed) [reply src/response/usage_report.rs:71]: `is_limited` is defined (for me, anyway) as "can't hit again".

**Review by davidor — APPROVED** [review]

**Comment by unleashed** [comment]
bors r=@davidor

**Pull request merged**